packages/api/cms-api/src/page-tree/blocks/createInternalLinkBlock.ts

```diff
@@ -0,0 +1,155 @@
+import {
+    AnnotationBlockMeta,
+    Block,
+    BlockData,
+    BlockField,
+    BlockIndexData,
+    BlockInput,
+    BlockMetaField,
+    BlockMetaFieldKind,
+    createBlock,
+    inputToData,
+    TransformResponse,
+} from "@comet/blocks-api";
+import { IsOptional, IsString, IsUUID } from "class-validator";
+
+import { PageTreeReadApi } from "../page-tree-read-api";
+import { PageExists } from "../validators/page-exists.validator";
+
+interface InternalLinkBlockTransformResponse extends TransformResponse {
+    targetPage: {
+        id: string;
+        name: string;
+        path: string;
+        documentType: string;
+    } | null;
+    targetPageAnchor?: string;
+}
+
+interface InternalLinkBlockDataInterface extends BlockData {
+    targetPageId?: string;
+    targetPageAnchor?: string;
+}
+
+interface InternalLinkBlockInputInterface extends BlockInput {
+    targetPageId?: string;
+    targetPageAnchor?: string;
+    transformToBlockData: () => InternalLinkBlockDataInterface;
+}
+
+interface CreateInternalLinkBlockOptions {
+    pageTreeNodeEntityName: string;
+}
+
+export function createInternalLinkBlock({
+    pageTreeNodeEntityName,
```

Comment:
Should we also export a default implementation with the default entity name "PageTreeNode" which will be used in 99% of all use cases? Would easy upgrading?